### PR TITLE
SW-4832 Draft planting sites

### PR DESCRIPTION
**Pull request by sgrimm**
Add CRUD endpoints for work-in-progress drafts of planting sites so clients can
allow users to save their work and continue later.

Drafts are visible to organization users with Owner, Admin, or Manager roles, but
can only be written by Owner and Admin users.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1670** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1670?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #1671** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1671?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1670?utm_source=stack-comment).

File: src/main/kotlin/com/terraformation/backend/tracking/db/DraftPlantingSiteStore.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            modifiedTime = now,
+            name = name,
+            numPlantingSubzones = numPlantingSubzones,
+            numPlantingZones = numPlantingZones,
+            organizationId = organizationId,
+            projectId = projectId,
+            timeZone = timeZone,
+        )
+
+    dslContext.attach(record)
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/tracking/db/DraftPlantingSiteStore.kt:73]: just curious, is this only needed when creating new records?

File: src/main/kotlin/com/terraformation/backend/tracking/api/DraftPlantingSitesController.kt [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    val numPlantingSubzones: Int?,
+    @Schema(
+        description =
+            "If the user has started defining planting zones, the number of zones defined so far.")
+    val numPlantingZones: Int?,
+    @Schema(description = "If the draft is associated with a project, its ID.")
+    val projectId: ProjectId?,
+    val timeZone: ZoneId?,
+) {
+  constructor(
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/tracking/api/DraftPlantingSitesController.kt:110]: stick the org id as well?

**Review by karthikbtf — APPROVED** [review]
This is great, very straightforward, leveraging jsonb definitely simplifies things.
One request in the feedback.

File: src/main/kotlin/com/terraformation/backend/tracking/db/DraftPlantingSiteStore.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/tracking/db/DraftPlantingSiteStore.kt:73]: Yeah. I'm kind of trying this pattern on for size. The `DAO` classes and the POJOs (the classes that have the `Row` suffix on their names in our setup) are kind of second-class citizens in jOOQ and the official recommendation is now to use `Record` classes for most purposes instead. They use a pattern more like ActiveRecord where you tell them to persist themselves using a persistence configuration (which is what's being set by the `attach` method).

**Review by karthikbtf — APPROVED** [review]
Thanks!!

**Comment by sgrimm** [comment]
### Merge activity

* **Feb 9, 5:45 PM EST**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1670).
* **Feb 9, 5:45 PM EST**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1670).

**Pull request merged**